spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -673,14 +674,14 @@ public void testSingleCommitWithCommitFailure() {
     RewriteDataFilesCommitManager util = spy(new RewriteDataFilesCommitManager(table));
 
     // Fail to commit
-    doThrow(new RuntimeException("Commit Failure")).when(util).commitFileGroups(any());
+    doThrow(new CommitFailedException("Commit Failure")).when(util).commitFileGroups(any());
 
     doReturn(util).when(spyRewrite).commitManager(table.currentSnapshot().snapshotId());
 
     assertThatThrownBy(() -> spyRewrite.execute())
         .as("Should fail entire rewrite if commit fails")
         .isInstanceOf(RuntimeException.class)
-        .hasMessage("Commit Failure");
+        .hasMessageContaining("Cannot commit rewrite");
```

Comment:
Why did this error message change? The exception message wasn't changed above.

Reply:
I updated this test to fail with `CommitFailedException` which is cleanable since the test prior to this change had an expectation that orphans were removed. When CommitFailedException gets thrown though, it ends up getting surfaced as a RuntimeException https://github.com/apache/iceberg/blob/main/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java#L317 with this error message "Cannot commit rewrite...". I've also just added another test to explicitly verify the non-cleanable case for rewriting data files.